service/builder/exporters_builder.go

```diff
@@ -88,10 +88,19 @@ func (exps Exporters) StartAll(logger *zap.Logger, host component.Host) error {
 }
 
 // ShutdownAll stops all exporters.
-func (exps Exporters) ShutdownAll() {
+func (exps Exporters) ShutdownAll() error {
+	var errs []error
 	for _, exp := range exps {
-		exp.Shutdown()
+		err := exp.Shutdown()
```

Comment:
Agreed. I can send a separate PR with that change, since it will require adding `sync.WaitGroup` and running shutdown in goroutines, not just a mechanical change.

Reply:
I agree, let's do it separately.

Technically the entire pipelines can be shutdown concurrently, but we need to carefully track exporters that are attached to the same pipeline to only shutdown once. Likewise for receivers attached to the same pipeline.